### PR TITLE
Event CRUD APIs

### PR DESCRIPTION
**Pull request by DiabolusGX**
Copy code
### Sample API payload

<table>
  <tr>
    <th>endpoint</th>
    <th>payload</th>
  </tr>

<tr>
<td>POST /event</td>
<td>

```json
{
    "name": "test",
    "description": "this is test description",
    "timestamp": 1719280440,
    "event_type": 1,
    "start_time": "2024-07-10T15:04:05Z",
    "min_robins": 5,
    "max_robins": 50,
    "event_location": {
        "name": "my place",
        "latitude": 123,
        "longitude": 123
    }
}
```

</td>
</tr>

<tr>
<td>GET /event/:event_id</td>
<td>

```json
{
    "status": {
        "status": "success",
        "message": "Event fetched successfully"
    },
    "event": {
        "event_id": "JdH2ndenPBUO",
        "name": "test",
        "description": "this is test description",
        "start_time": "2024-07-10T15:04:05Z",
        "event_type": 1,
        "event_location": {
            "location_id": 5,
            "name": "my place",
            "latitude": 123,
            "longitude": 123
        },
        "min_robins": 5,
        "max_robins": 50,
        "created_at": "2024-07-09T09:52:40Z",
        "updated_at": "0001-01-01T00:00:00Z"
    }
}
```

</td>
</tr>

</table>

### CODE REVIEW

**Review by gunjanpatidar — COMMENTED** [review]
Looks good to me

**Comment by vishal91** [comment]
I think we should add an ORM. https://gorm.io/index.html 

Will help us a lot.

**Comment by DiabolusGX** [comment]
Added gorm, PR is ready for review from my side.

**Review by shubhsherl — APPROVED** [review]

**Comment by shubhsherl** [comment]
we might also need event photo, end_datetime, status and coordinator details
@DiabolusGX 

(from event page)
Screenshot 2024-07-02 at 1 25 38 AM

(from home page)
Screenshot 2024-07-02 at 1 37 00 AM

File: internal/repositories/sql/event/sql_event_repository.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+// NewEventRepository creates a new EventRepository
+func NewEventRepository(logger *zap.Logger, db *database.SqlDB) *EventRepository {
+	return &EventRepository{logger, db}
+}
+
+// CreateEvent creates a new event within a transaction
+func (r *EventRepository) CreateEvent(req *eventModel.CreateEventRequest) (*eventModel.CreateEventResponse, error) {
+	id, err := nanoid.GetID()
```
Comment (shubhsherl) [inline internal/repositories/sql/event/sql_event_repository.go:32]: should we use uuid.UUID ?
github.com/google/uuid
uuid.NewRandom()

Comment (DiabolusGX) [reply internal/repositories/sql/event/sql_event_repository.go:32]: those are larger ids, mostly un-readable so referring to this - https://github.com/RobinHoodArmyHQ/robin-api/pull/7#discussion_r1666205987
used nanoids instead - https://zelark.github.io/nano-id-cc/

**Review by gunjanpatidar — APPROVED** [review]

**Pull request merged**